File: app/src/main/java/com/example/miwok/WordAdapter.java
package com.example.miwok;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;

import java.util.ArrayList;
import java.util.List;

public class WordAdapter extends ArrayAdapter<Word> {

    private int colorid;

    public WordAdapter(Context context, ArrayList<Word> words, int ColorId) {
        super(context, 0, words);
        colorid = ColorId;
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View listview = convertView;
        if (listview == null) {
            listview = LayoutInflater.from(getContext()).inflate(R.layout.list_item, parent, false);
        }

        Word word = getItem(position);

        TextView textView = listview.findViewById(R.id.miwokWord);
        textView.setText(word.getMiwok());

        TextView textView1 = listview.findViewById(R.id.engWord);
        textView1.setText(word.getEnglish());

        ImageView imageView = listview.findViewById(R.id.imageid);
        if (word.hasImage()){
            imageView.setImageResource(word.getImageid());
            imageView.setVisibility(View.VISIBLE);
        } else {
            imageView.setVisibility(View.GONE);
        }

        View container = listview.findViewById(R.id.container);
        int color = ContextCompat.getColor(getContext(), colorid);
        container.setBackgroundColor(color);

        return listview;
    }
}

File: app/src/main/java/com/example/miwok/WordAdapter.kt
package com.example.miwok

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat

class WordAdapter(context: Context, words: List<Word>, private val colorId: Int) :
    ArrayAdapter<Word>(
        context, 0, words
    ) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val view =
            convertView ?: LayoutInflater.from(context).inflate(R.layout.list_item, parent, false)

        val word = getItem(position)!!

        view.findViewById<TextView>(R.id.miwokWord).text = context.getString(word.mMiwok)

        view.findViewById<TextView>(R.id.engWord).text = context.getString(word.mEng)

        val imageView = view.findViewById<ImageView>(R.id.imageid)

        if (word.hasImage()) {
            imageView.setImageResource(word.mImageid!!)
            imageView.visibility = View.VISIBLE
        } else {
            imageView.visibility = View.GONE
        }

        view.findViewById<View>(R.id.container)
            .setBackgroundColor(ContextCompat.getColor(context, colorId))

        return view
    }
}
